api/src/main/java/com/spotify/metrics/core/MetricId.java

```diff
@@ -123,7 +123,7 @@ private String extendKey(final String part) {
     public MetricId tagged(Map<String, String> add) {
         final TreeMap<String, String> tags = new TreeMap<>(this.tags);
         tags.putAll(add);
-        return new MetricId(key, tags);
+        return new MetricId(key, Collections.unmodifiableSortedMap(tags));
```

Comment:
how about wrapping the tags in an unmodifiable map in the `MetricId` constructor to make it totally consistent and future-proof?

Reply:
Done!